lib/galaxy/model/migrations/alembic/versions_gxy/b182f655505f_add_workflow_source_metadata_column.py

```diff
@@ -20,12 +20,16 @@
 branch_labels = None
 depends_on = None
 
+# database object names user in this revision
+table_name = "workflow"
+column_name = "source_metadata"
+
 
 def upgrade():
-    table, column = "workflow", "source_metadata"
-    with ignore_add_column_error(table, column):
-        op.add_column(table, Column(column, JSONType))
+    if not column_exists(table_name, column_name):
+        op.add_column(table_name, Column(column_name, JSONType))
 
 
 def downgrade():
-    drop_column("workflow", "source_metadata")
+    if column_exists(table_name, column_name):
```

Comment:
Do we need the check here? If we wanted to the migration to be "symmetrical" I suppose it should be a no-op. But given that the sqlalchemy-migrate migrations don't really fail I think we can drop unconditionally.

Reply:
I don't think we need it here, but maybe I'm wrong. Here's my reasoning: this would be executed only by alembic, and that means alembic *has been used to upgrade* to the version containing that column - so the column *must* exist. The revision script will fail if the column does *not* exists - which indicates that the database is in an inconsistent state, which is when, I think, we want it to fail loud and fast. So, it's non-symmetrical by design, at least following this logic.

Reply:
> The revision script will fail if the column does _not_ exists

I agree with what you said before this, but I don't see how the script will fail. Don't you need to remove the if statement here ?